### PR TITLE
DOCSP-4485: Add email field to the feedback widget

### PR DESCRIPTION
**Pull request by sophstad**
[[JIRA](https://jira.mongodb.org/browse/DOCSP-4485)] This PR:
- Adds an input field for email to the feedback widget
- Does not require users to enter an email
- If text is entered in the field, validation is performed to ensure the email appears valid
- If the email is invalid, form submission is blocked and an error is shown

screen shot 2019-02-21 at 1 25 14 pm
screen shot 2019-02-21 at 1 25 23 pm
screen shot 2019-02-21 at 1 25 35 pm

### CODE REVIEW

File: themes/mongodb/src/widgets/deluge/InputField.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+class InputField extends preact.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            'error': false,
+            'text': ''
+        };
+
+        this.handleChange = this.handleChange.bind(this);
```
Comment (i80and) [inline themes/mongodb/src/widgets/deluge/InputField.js]: I'd use a class field for `handleChange`

File: themes/mongodb/src/css/feedback.css [hunk 2]
```diff
@@ -43,6 +43,10 @@
     width: 100%; }
   .deluge textarea:invalid {
     color: red; }
+  .deluge input {
+    width: 100%; }
+  .deluge input:invalid {
+    color: red; }
```
Comment (i80and) [inline themes/mongodb/src/css/feedback.css]: We should replace these `red`'s with `var(--tomato)`.

I think it's also worth combining these invalid states:
```
.deluge textarea:invalid,
.deluge input:invalid {
  color: var(--tomato);
}
```

**Comment by sophstad** [comment]
@i80and opened this PR a bit too soon—Gilberto is having a meeting tomorrow to discuss updates to the feedback widget, so there may be some changes incoming. But thanks for the preliminary review!

**Comment by sophstad** [comment]
image
Updated to include a prompt message

**Comment by i80and** [comment]
@sophstad could you rebase please?

**Pull request merged**